starlite/contrib/sqlalchemy/base.py

```diff
@@ -0,0 +1,112 @@
+"""Application ORM configuration."""
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, TypeVar
+from uuid import UUID, uuid4
+
+from pydantic import AnyHttpUrl, AnyUrl, EmailStr
+from sqlalchemy import JSON, MetaData, String, Uuid
+from sqlalchemy.event import listens_for
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    Session,
+    declarative_mixin,
+    declared_attr,
+    mapped_column,
+    registry,
+)
+
+BaseT = TypeVar("BaseT", bound="Base")
+
+convention = {
+    "ix": "ix_%(column_0_label)s",
+    "uq": "uq_%(table_name)s_%(column_0_name)s",
+    "ck": "ck_%(table_name)s_%(constraint_name)s",
+    "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+    "pk": "pk_%(table_name)s",
+}
+"""Templates for automated constraint name generation."""
+
+
+@listens_for(Session, "before_flush")
+def touch_updated_timestamp(session: Session, *_: Any) -> None:
```

Comment:
I think we should find a way to only attach this listener only if the audit base is used, or make it a thing that is configurable by the end user.

Reply:
also, on some backends this can be done using server side default, e.g., mysql/mariadb.

Reply:
I didn't do this because it's not as easy to detect if the session is dirty.  A `server_onupdate` fires any time an update happens, even if there is no actual modification.  However, i do like the idea of server generated defaults for timestamps.

Reply:
Oh cool, well that is a plus for doing it client side!